core/src/main/scala/com/yahoo/maha/core/RequestModel.scala

```diff
@@ -158,6 +158,7 @@ case class RequestModel(cube: String
                         , requestedFkAliasToPublicDimensionMap: Map[String, PublicDimension]
                         , orFilterMeta: Set[OrFilterMeta]
                         , dimensionRelations: DimensionRelations
+                        , useNewFeature: Boolean = false
```

Comment:
it is not used anywhere. I think instead of flag, we can use Map to Store feature flags. 
Create a trait called MahaFeature and you can use map of MahaFeature-> Boolean